tests/Constraints/PropertyNumberTest.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+/*
+ * This file is part of the JsonSchema package.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace JsonSchema\Tests\Constraints;
+
+class PropertyNumberTest extends BaseTestCase
+{
+    protected $validateSchema = true;
+
+    public function getInvalidTests(): array
+    {
+        return [];
+    }
+
+    public function getValidTests(): array
+    {
+        return [
+            [
+                '{"some_property":{"some_sub_property":{"4":10,"2":10}}}',
+                '{
```

Comment:
This test seems needlessly complicated - what is it that you are actually intending to test? Tests should test one thing, clearly. If there is a need to test multiple things, there should be a clear test for each thing.

Reply:
This is an example from my case. As I mentioned before, unfortunately, I can not look into that more closely and trace the root cause at the moment.

Hope to come back to this later or maybe someone else jumps into the same problem and find some time to dig it.

Reply:
As it stands currently, it's not an appropriate test. Tests need to have a clear purpose; they can't just be a paste of "this is a piece of my project that I think should pass validation".

> I can not look into that more closely and trace the root cause at the moment.

If this relates to the error reported in your original comment, the root cause is as the error says: you are attempting to use an integer as a JSON property name, which is in violation of the spec. You can't do that.

JSON has no concept of arrays that use arbitrary keys (i.e. associative arrays). You cannot legitimately use an array with a non-zero basis, or with index holes, because that is by definition an associative array. If you want to validate it, you'll need to turn it into a JSON-compliant object.

Reply:
> As it stands currently, it's not an appropriate test. Tests need to have a clear purpose; they can't just be a paste of "this is a piece of my project that I think should pass validation".

That's right.

> you are attempting to use an integer as a JSON property name, which is in violation of the spec. You can't do that.

The JSON property is a numeric string. e.g. `"4"` or `"2"` in `'{"some_property":{"some_sub_property":{"4":10,"2":10}}}'`. To my knowledge it is a perfectly valid JSON. Is there a particular spec rule which states that a field cannot be made out of 0-9 characters?

Reply:
A string containing a number is fine - that's still a string, which is what matters.

My comment about using integers as property names relates to your original post in this PR: using integers as property names is what you said that you are doing to trigger the error. I wasn't referring to the test content.

Reply:
Initially I though that integer keys of an array caused the issue in the domain model. However the model is normalized into an array before it is passed to JSON validator and the integer keys are already casted into regular strings. So at the end of the day, **validator is not able to handle JSON fields which are numeric strings.**

This was a dirty workaround, so I made a PR hoping me or someone else could come to this later.

I've tried to simplify the test input, but was unable to do it quickly.

Reply:
> validator is not able to handle JSON fields which are numeric strings.

This seems extremely odd - but would definitely be a bug if it is the case. I will investigate that today.

Is there any particular situation which is required to trigger it, or is it simply any property with a numeric string as the name?

Reply:
I found a more simple example. I'll push it soon.